### PR TITLE
Let "Loading terrain" close screen only when needed

### PR DESCRIPTION
**Pull request by bluegreensea**
this let Immediate Respawn works like before 1.20.3 (keep chat screen, etc) but more
when cross dimension (respawn or tp to a different dimension) keep screen too

### CODE REVIEW

**Comment by kennytv** [comment]
Thanks for the PR, can you rephrase/go into more detail on what this fixes? also please spam me if I don't get to this by 1.21

**Comment by bluegreensea** [comment]
Here are before and after video

https://github.com/kennytv/kennytvs-epic-force-close-loading-screen-mod-for-fabric/assets/35960465/dfdd21bf-83d1-4a1b-a382-9dc5071d05a5

https://github.com/kennytv/kennytvs-epic-force-close-loading-screen-mod-for-fabric/assets/35960465/a8709255-df83-4d75-bce8-ca648e02a519

**Comment by bluegreensea** [comment]
It doesn't work with reconfiguration and strangely doesn't work on paper (idk why, needs more investigation)

**Review by kennytv — APPROVED** [review]
Slightly late, but thanks!
